### PR TITLE
React 18 upgrade

### PR DESCRIPTION
**Pull request by mgcm**
**:heavy_check_mark: Checklist**

- [ ] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [X] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [X] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: packages/react-sdk/src/state/useWhiteboardManager.test.tsx [hunk 1]
```diff
@@ -50,10 +50,12 @@ describe('useWhiteboardManager', () => {
   });
 
   it('hook should throw without context', () => {
-    const { result } = renderHook(() => useWhiteboardManager());
+    jest.spyOn(console, 'error').mockImplementation(() => {});
```
Comment (weeman1337) [inline packages/react-sdk/src/state/useWhiteboardManager.test.tsx:53]: Shouldn't there be a `mockRestore` after each test?

(for all mocked console.*)

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Thanks for the hard work to get all tests running again :+1: 

In most places with an `act` there may be a more elegant solution. But let's get back to this at a later time.

The only problematic thing I could find was the console mock without a reset (see comment).

**Review by weeman1337 — DISMISSED** [review]

File: packages/react-sdk/src/state/useWhiteboardManager.test.tsx (hunk 1 above)
Comment (mgcm) [reply packages/react-sdk/src/state/useWhiteboardManager.test.tsx:53]: of course there should be! 👯‍♂️👯‍♂️ I've updated all occurrences...

**Review by weeman1337 — APPROVED** [review]

**Review by weeman1337 — DISMISSED** [review]

**Comment by mgcm** [comment]
sorry, need _yet_ another approval - the previous released updated the neoboard package version...

**Comment by mgcm** [comment]
@weeman1337 sorry, I need yet another approval, the previous release updated the neoboard react-sdk package dependency...

**Review by weeman1337 — APPROVED** [review]

**Pull request merged**